src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -252,6 +237,52 @@ internal void SetMaxMetricPointsPerMetricStream(int maxMetricPointsPerMetricStre
         }
     }
 
+    private static void DeactivateMetric(Metric metric)
+    {
+        metric.InstrumentDisposed = true;
+
+        var metricStreamIdentity = metric.InstrumentIdentity;
+
+        OpenTelemetrySdkEventSource.Log.MetricInstrumentDeactivated(
```

Comment:
I think we should log this in the `GetMetricsBatch` method after `this.metrics[i] = null;`. That is when the SDK disables further processing/export of the metric.

Reply:
So... working through this I found a race condition. If an instrument is deactivated and then activated before an export has a chance to run, the next export removes it and then all data is dropped. Yikes!

Here's the fix I did: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4958/commits/f73d309c98ec73a29bc48f7d05463d2d19fad2ef

There are now 3 events: 1 for deactivate, 1 for activate, and 1 for the actual cleanup/removal during export.

Reply:
Yes, this is a known issue #4384. We never prioritized fixing this as it was a very niche use case up until now.